src/Service/OrderHelper.php

```diff
@@ -0,0 +1,61 @@
+<?php declare(strict_types=1);
+
+namespace MoptWorldline\Service;
+
+use MoptWorldline\Bootstrap\Form;
+use Shopware\Core\Checkout\Order\OrderEntity;
+use Shopware\Core\Checkout\Payment\Exception\InvalidTransactionException;
+use Shopware\Core\Framework\Context;
+use Shopware\Core\Framework\DataAbstractionLayer\EntityRepository;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Criteria;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Filter\EqualsFilter;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Filter\MultiFilter;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Filter\NotFilter;
+
+class OrderHelper
+{
+    /**
+     * @param Context $context
+     * @param EntityRepository $orderRepository
+     * @param string $hostedCheckoutId
+     * @return OrderEntity|null
+     */
+    public static function getOrder(
+        Context                   $context,
+        EntityRepository $orderRepository,
+        string                    $hostedCheckoutId
+    ): ?OrderEntity
+    {
+        $criteria = new Criteria();
+        $criteria->addAssociation('transactions');
+        $criteria->addFilter(
+            new MultiFilter(
+                MultiFilter::CONNECTION_AND,
+                [
+                    new EqualsFilter(
+                        \sprintf('customFields.%s', Form::CUSTOM_FIELD_WORLDLINE_PAYMENT_HOSTED_CHECKOUT_ID),
+                        $hostedCheckoutId
+                    ),
+                    new NotFilter(
+                        NotFilter::CONNECTION_AND,
+                        [
+                            new EqualsFilter(
+                                \sprintf('customFields.%s', Form::CUSTOM_FIELD_WORLDLINE_PAYMENT_HOSTED_CHECKOUT_ID),
+                                null
+                            ),
+                        ]
+                    ),
+                ]
+            )
+        );
+
+        /** @var OrderEntity|null $order */
+        $order = $orderRepository->search($criteria, $context)->getEntities()->first();
+
+        if ($order === null) {
+            throw new InvalidTransactionException('');
```

Comment:
Why no message?

Reply:
Standard message is 'The transaction with id {{ transactionId }} is invalid or could not be found.'
At this point we have only external transaction ID, providing it here no make sense.

We will face this exception on Worldline malfunction or wrong return URL config, so external transaction ID will not help.
But I'll add it to log in refactoring merge request.